recipes/default.rb

```diff
@@ -62,13 +63,21 @@
   else
     rpc_namenode_fqdn = private_recipe_ip("hops", "nn")
   end
+
+  if node['hops']['rm']['private_ips'].include?(my_ip)	
+    resourcemanager_fqdn = my_ip;
+  else
+    resourcemanager_fqdn = private_recipe_ip("hops","rm")	
+  end
   zookeeper_fqdn = private_recipe_ip('kzookeeper', 'default')
 end
 
-if node['hops']['rm']['private_ips'].include?(my_ip)	
-  resourcemanager_fqdn = my_ip;
-else
-  resourcemanager_fqdn = private_recipe_ip("hops","rm")	
+if exists_local("hops", "rm")
```

Comment:
What would be the `resourcemanager_fqdn` in the Worker in this case?

Reply:
it depends if service discovery is enabled it would `resourcemanger.service.consul` otherwise it will be the ip